### DISCUSSION TITLE
Open Map Tiles Style Sheet

### DISCUSSION THREAD

**Discussion by mdoakes42** [onthegomap/planetiler#746] [General]
I am trying to utilize the OpenMapTiles Style Sheets in Geoserver.  Most of the styles are looking for name:latin and/or name:nonlatin.  "text-field": "{name:latin} {name:nonlatin}",     However, When I generate the vector tiles using the OsmMapTiles profile (or the standard/default profile)..   Neither appears to have those fields in the mbtiles and thus I'm getting errors about these fields not existing.   Anyone experience this issue and have a work around?    Are these fields part of the standard schema?  If so why aren't thye exported in the mbtiles?  Thanks

**Comment by mdoakes42** [comment]
Also wondering why the field brunnel does not show up in the "water" layer.   This is in the standard schema.    I see a note about brunnel not being included in the transportation layer.. But it should still be included here?  Any reason why this might happen. I'm using the docker container openmaptiles/planetiler-openmaptiles:latest  to run this. Thanks

**Reply by msbarry** [reply]
Do you have any specific example areas/features?  There is [code in the openmaptiles profile to set brunnel](https://github.com/openmaptiles/planetiler-openmaptiles/blob/0f900ea2b6d25d54ff88fb485c0081a845bf6b80/src/main/java/org/openmaptiles/layers/Water.java#L122)   and also [`name:latin` and `name:nonlatin`](https://github.com/openmaptiles/planetiler-openmaptiles/blob/0f900ea2b6d25d54ff88fb485c0081a845bf6b80/src/main/java/org/openmaptiles/util/OmtLanguageUtils.java#L101-L102) - so you might have found a bug preventing them from getting set in certain situations and we'd need more details to debug.

**Reply by mdoakes42** [reply]
Thank you, I'll take a look.  Similarly,   There are fields that are defined in the openmaptiles schema (and included in their styles.json files) but are not in the output.mbtiles file.  A couple examples

POI Layer - The rank field is not included
Boundary Layer - claimed_by is missing.

Is this a bug?  or is this intentional?     Thanks

**Comment by mdoakes42** [comment]
I think I might see whats happening.   When I did a test with "monaco" compared with "egypt" the outputed fields were different.   It looks like I only get the fields available from what the source data contains.  (IE,  There is NOT a standard schema output.. It is more of what is contained in the source data set).   

As an example when I dumped egypt I got the aerodrome_label layer. (that wasn't even included in the monaco one).

Also,  Starting with boundary for egypt I see extra fields such as adm0_r, disputed_name, adm0_l
Building layer also has a "hide_3d" in egypt which monaco doesn't have.

My guess is if I dump the "planet" file I will see ALL the fields/schema.       I think this might be why I was having an issue with fields missing.  It was because they didn't exist in the small source that I was querying.     

Is this something you can confirm?  Thanks

**Reply by msbarry** [reply]
Oh are you talking about the tilejson metadata? That only contains the attributes present on features in the actual tile output.

**Comment by mdoakes42** [comment]
brunnel also shows up now with the egypt process.

I think this was just a matter of understanding how the schema's work and that it could be different depending on what fields are available in that source pbf.

**Comment by mdoakes42** [comment]
I did a full planet dump and the "rank" field is missing in the POI layer.  Any ideas why?   I'm running this by using docker 

docker run -e JAVA_TOOL_OPTIONS="-Xmx64g" -v "$(pwd)/data":/data openmaptiles/planetiler-openmaptiles:latest --osm-path=/data/sources/planet.osm.pbf.............

**Reply by msbarry** [reply]
Ah I see what's happening. The tilejson attribute metadata is computed during source feature processing, but rank gets added during tile post-processing. So rank is in the output, but it's missing from the tilejson metadata. I opened #749 to move computing the tilejson metadata to when the tiles are emitted instead to account for attributes added/removed during post-processing.

**Reply by mdoakes42** [reply]
Thank you Michael

**Reply by msbarry** [reply]
Fixing in #752
